### DISCUSSION TITLE
Wrong task instance available in context of on_success_callback

### DISCUSSION THREAD

**Discussion by harthur** [apache/airflow#19371] [Q&A] [answered]
When you use `context[task_instance']` in an `on_success_callback`, you don't  get the "right" task instance. In particular, it's not a reference to the same task instance that was actually used during execution. It's some shell of the same type of task.

In Airflow 1, we were successfully using the context's task instance to get the `job_id` of our AWS Batch operator and handily print the link the that job's Batch logs from Slack. At the time, we found that `context['task']` had a similar incorrect shell object, and had to use `context['task_instance'].task` to get the real one.

Is there any way to get the actual task instance or task used during execution, from the `on_success_callback`? I have also tried using `context['dag'].tasks`, but those are also the wrong objects.

**Comment by potiuk** [comment, marked as answer]
It's been fixed in 2.2.0 https://github.com/apache/airflow/pull/18516/files

**Reply by harthur** [reply]
Thank you!

**Reply by f0cus10** [reply]
Hi, we're using Airflow `v2.2.2` but am encountering this issue.

When accessing `context['ti']` on `on_failure_callback`, the first task that is NOT a `DummyOperator` is returned.

**Reply by potiuk** [reply]
Can you open an issue about it. please - with all details. It's amazingly counter-productive to raise a new issue which is similar - but likely differnt - as a comment to a discussion that has been marked as answered. 

If you wish to simply complain about - that's fine, but if you want anything done to fix it - please provide all the details and spend time on reporting a quality issue with enough information to be able to act on it @f0cus10. If you don't want anything done - yep, you just achieved it.
